code/modules/food_and_drinks/food/snacks_pie.dm

```diff
@@ -26,24 +26,28 @@
 	tastes = list("pie" = 1)
 
 /obj/item/weapon/reagent_containers/food/snacks/pie/cream/throw_impact(atom/hit_atom)
-	if(!..()) //was it caught by a mob?
-		var/turf/T = get_turf(hit_atom)
-		new/obj/effect/decal/cleanable/pie_smudge(T)
-		reagents.reaction(hit_atom, TOUCH)
-
-		if(ishuman(hit_atom))
-			var/mob/living/carbon/human/H = hit_atom
-			var/mutable_appearance/creamoverlay = mutable_appearance('icons/effects/creampie.dmi')
-			if(H.dna.species.id == "lizard")
-				creamoverlay.icon_state = "creampie_lizard"
-			else
-				creamoverlay.icon_state = "creampie_human"
-			H.Knockdown(20) //splat!
-			H.adjust_blurriness(1)
-			visible_message("<span class='userdanger'>[H] was creamed by [src]!!</span>")
-			H.add_overlay(creamoverlay)
-
-		qdel(src)
+	. = ..()
+	addtimer(CALLBACK(src, .proc/splat, hit_atom), 0.5)
```

Comment:
err wait yeah why does it impact before hitby checks if it was caught, that's bizzare

I reckon refactor it so that there is 
```
item.on_throw_encounter(target) //we've collided with a mob in world space, but not yet hit it
if(target.hitby(item)) //have we hit + plus deal mob damage
   item.throw_impact(target) //post hit behaviour for the item
```